lighthouse-core/lib/dependency-graph/simulator/network-analyzer.js

```diff
@@ -385,6 +385,54 @@ class NetworkAnalyzer {
     return NetworkAnalyzer.summarize(estimatesByOrigin);
   }
 
+
+  /**
+   * Computes the average throughput for the given records in bits/second.
+   * Excludes data URI, failed or otherwise incomplete, and cached requests.
+   * Returns Infinity if there were no analyzable network records.
+   *
+   * @param {Array<LH.Artifacts.NetworkRequest>} networkRecords
+   * @return {number}
+   */
+  static estimateThroughput(networkRecords) {
+    let totalBytes = 0;
+    const timeBoundaries = networkRecords.reduce((boundaries, record) => {
+      const scheme = record.parsedURL && record.parsedURL.scheme;
+      if (scheme === 'data' || record.failed || !record.finished ||
+          record.statusCode > 300 || !record.transferSize) {
+        return boundaries;
+      }
+
+      totalBytes += record.transferSize;
+      boundaries.push({time: record.responseReceivedTime, isStart: true});
+      boundaries.push({time: record.endTime, isStart: false});
+      return boundaries;
+    }, /** @type {Array<{time: number, isStart: boolean}>} */([])).sort((a, b) => a.time - b.time);
+
+    if (!timeBoundaries.length) {
+      return Infinity;
+    }
+
+    let inflight = 0;
+    let currentStart = 0;
+    let totalDuration = 0;
+    timeBoundaries.forEach(boundary => {
+      if (boundary.isStart) {
+        if (inflight === 0) {
+          currentStart = boundary.time;
+        }
+        inflight++;
+      } else {
+        inflight--;
+        if (inflight === 0) {
+          totalDuration += boundary.time - currentStart;
+        }
+      }
+    });
+
+    return totalBytes * 8 / totalDuration;
```

Comment:
Historically it was measured in bytes (which is why all the tests are now `* 8`), but because this PR moves the code into network throughput where everything is measured in bits I wanted to make it consistent. Since I was just doing a straight copy with minimal changes and network records use bytes, I thought it was most straightforward to do it at the end but could also do a `totalBytes -> totalBits` rename and do it as it sums if you think that's clearer :)

Reply:
I think to me it would be clearer if it collected the `totalBits` and then return that, but then that's a lot more multiplication for very little reason.  It's fine like this if the standard is to return bits, I just thought it was funny to calculate the bits then just do bytes * 8 for the tests.